SCA: Add Parasoft C++test support

yashi opened the pull request:
This PR adds Parasoft C++test as one of the [Static Code Analysis (SCA)](https://docs.zephyrproject.org/latest/develop/sca/index.html) tools.

This fixes #67816.

yashi committed:
cmake: sca: Add Parasoft C++test

This commit adds Parasoft C++test as a Static Analyser using Zephyr's SCA
framework.

By specifing -DZEPHYR_SCA_VARIANT=cpptest to west build, a cpptestscan.bdf
file will be generated under builddir/sca/cpptest/.

Signed-off-by: Yasushi SHOJI <yashi@spacecubics.com>

tejlmand commented on doc/develop/sca/cpptest.rst:
```diff
@@ -0,0 +1,40 @@
+########################
+
+Parasoft `C++test <https://www.parasoft.com/products/parasoft-c-ctest/>`__ is a software testing and
+static analysis tool for C and C++. It is a commercial software and you must aquire a commercial
```
```suggestion
static analysis tool for C and C++. It is a commercial software and you must acquire a commercial
```

yashi replied:
Fixed. Thanks.

tejlmand commented on doc/develop/sca/cpptest.rst:
```diff
@@ -0,0 +1,40 @@
+
+Please refer to Parasoft C++test documentation for more details.
+
+To import and generate a report file, something like the follwoing should work.
```
```suggestion
To import and generate a report file, something like the following should work.
```

yashi replied:
Fixed. Thanks.

tejlmand reviewed (changes requested):
looks good.

Couple of minor spelling mistakes to be fixed.

kartben commented on cmake/sca/cpptest/sca.cmake:
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2024, Space Cubics, LLC.
+
+find_program(CPPTESTSCAN cpptestscan REQUIRED)
+message(STATUS "Found Parasoft C++test: ${CPPTESTSCAN}")
```
looks like the official name of the product is "Parasoft C/C++test" so it would make sense to stick to this across the entire PR (unless Parasot C++test is a thing and I just missed it)

yashi replied:
Thanks.  Fixed.

Would it be nice if the line reads something like: `Found SCA: Parasoft C/C++test (/usr/bin/cpptestscan)`? 

- It aligns with other "Found" messages
- It tells what the Parasoft C/C++test is by the string `SCA`

If you like it, I can add another commit to chnage the lines from codechecker and sparse.

yashi replied:
I've added a commit to add "SCA" string for other SCA tools.  If you don't like it, please drop the commit.  Thanks.

yashi committed:
cmake: sca: Add "SCA" to the Found messages

This commit enhances the clarity of the build log by adding the "SCA"
prefix to the "Found" messages generated by sca/*/sca.cmake. This change
improves the readability of the `west build` log for users who may not be
familiar with these tools, providing more informative and understandable
output.

Signed-off-by: Yasushi SHOJI <yashi@spacecubics.com>

henrikbrixandersen commented:
@yashi Please rebase on `main` and resolve any merge conflicts.

yashi commented:
@henrikbrixandersen Done

yashi committed:
doc: develop: sca: Add Parasoft C/C++test

Add documentation for Parasoft C/C++test for Static Code Analysis.

Signed-off-by: Yasushi SHOJI <yashi@spacecubics.com>

yashi commented:
@henrikbrixandersen Done
